### ISSUE TITLE
spacy.prefer_cpu()

### ISSUE THREAD

**Issue by carschno** [explosion/spaCy#5018]
## Feature description
It should be possible to deactivate GPU usage. 

Use case: multi-core (CPU) processing (e.g. with Joblib) throws an exception when GPU is enabled, see e.g. https://github.com/chainer/chainer/issues/1087

This could be implemented in a `spacy.prefer_cpu()` function, in order to undo `spacy.prefer_gpu()`/`spacy.require_gpu()`.

## Could the feature be a [custom component](https://spacy.io/usage/processing-pipelines#custom-components) or [spaCy plugin](https://spacy.io/universe)?

Not sure if this could be an extension.

**Comment by honnibal** [comment]
Yeah that's reasonable. Do you want to make a PR?

**Comment by carschno** [comment]
I'm not sure that this is supported by cupy though, because after call `device.use()`, there does not seem to be a way to disable the device:

> If you want to switch a device temporarily, use the with statement.

see https://docs-cupy.chainer.org/en/stable/reference/generated/cupy.cuda.Device.html#cupy.cuda.Device

At the same time, I do not see a way to use a `with` statement in SpaCy without major changes in the logic.

**Comment by adrianeboyd** [comment]
I've also wanted to have this, and I think for spacy you mainly have to tell thinc to go back to using numpy when it loads new models rather than cupy:

```
import spacy
spacy.require_gpu()
nlp_gpu = spacy.load("en_core_web_sm")
from thinc.neural._classes.model import Model
from thinc.neural.ops import NumpyOps, CupyOps
Model.Ops = NumpyOps
Model.ops = NumpyOps()
nlp_cpu = spacy.load("en_core_web_sm")
assert type(nlp_cpu.tagger.model.ops) == NumpyOps
assert type(nlp_gpu.tagger.model.ops) == CupyOps
```

This works for simple cases, but I'm not sure it will work in the more complicated `joblib` setup?

**Comment by honnibal** [comment]
This is kind of messy in the current version of Thinc; it's cleaner in Thinc 8.x. We hope to have a nightly up soon that uses the new Thinc.

With the new version of Thinc it's simply a matter of calling `set_current_ops`, which changes a global setting held by a contextvar. So this should behave correctly across threads and in coroutines.

**Comment by svlandeg** [comment]
The nightly took a little bit longer than we expected, but to compensate there are loads of new features in [the pre-release of spaCy v3](https://github.com/explosion/spaCy/releases/tag/v3.0.0rc1) ;-)

I think a method `spacy.require_cpu()` would make sense, and will delegate to `set_current_ops` as Matt suggested. I created a [quick PR](https://github.com/explosion/thinc/pull/409) in Thinc here, which we can then call from within spaCy.

**Comment by svlandeg** [comment]
`spacy.require_cpu()` will be available from v3.0.0rc3 onwards :-)
